test/e2e/metering_manual_install_tests.go

```diff
@@ -107,6 +112,105 @@ func testManualMeteringInstall(
 	assert.NoError(t, err, "capturing logs and uninstalling metering should produce no error")
 }
 
+func createNFSProvisioner(ctx *deployframework.DeployerCtx) error {
+	//Waiting for Metering pods
+	ctx.TargetPodsCount = nonHDFSTargetPodCount
+
+	_, err := createTestingNamespace(ctx.Client, "nfs")
+	if err != nil {
+		return err
+	}
+
+	files := map[string]string{
+		"pod":          "/test/testhelpers/nfs_server.yaml",
+		"service":      "/test/testhelpers/nfs_service.yaml",
+		"storageClass": "/test/testhelpers/nfs_storageclass.yaml",
+
+	}
+	server := &corev1.Pod{}
+	service := &corev1.Service{}
+	storageClass := &v1beta1.StorageClass{}
+
+	for name, file := range files {
+
+		switch name {
+		case "pod":
+			if err := deploy.DecodeYAMLManifestToObject(file, server); err != nil {
+				return err
+			}
+			ctx.Client.CoreV1().Pods("nfs").Create(context.TODO(), server, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		case "service":
+			if err := deploy.DecodeYAMLManifestToObject(file, service); err != nil {
+				return err
+			}
+			ctx.Client.CoreV1().Services("nfs").Create(context.TODO(), service, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		case "storageClass":
+			if err := deploy.DecodeYAMLManifestToObject(file, storageClass); err != nil {
+				return err
+			}
+			ctx.Client.StorageV1beta1().StorageClasses().Create(context.TODO(), storageClass, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		}
+	}
+	// wait and Poll , get service
+	svc, err := ctx.Client.CoreV1().Services("nfs").Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+	if apierrors.IsNotFound(err) && svc.Spec.ClusterIP == ""{
+
+		err = wait.Poll(3*time.Second, 5*time.Minute, func() (bool, error) {
+
+			svc, err = ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+
+			return svc.Spec.ClusterIP != "", err
+
+		})
+
+	}
+
+	nfsFile := "/test/testhelpers/nfs_persistentvolume.yaml"
+	nfs_pv := &corev1.PersistentVolume{}
+	nfs_pv.Spec.NFS.Server = svc.Spec.ClusterIP
+	//decode pvc
+	if err := deploy.DecodeYAMLManifestToObject(nfsFile, nfs_pv); err != nil {
+		return err
+	}
+	_, err = ctx.Client.CoreV1().PersistentVolumes().Create(context.TODO(), nfs_pv, metav1.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
+	}
+
+}// end of function
+
+func createTestingNamespace(client kubernetes.Interface, namespace string) (*corev1.Namespace, error) {
+	ns, err := client.CoreV1().Namespaces().Get(context.Background(), namespace, metav1.GetOptions{})
+	//Get(ctx context.Context, name string, opts metav1.GetOptions) (*v1.Namespace, error)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return ns, err
```

Comment:
I think we should just return `return nil, err` for better readability in the case the GET call failed and it failed in a manner we weren't expecting.